system-parachains/bridge-hubs/bridge-hub-polkadot/src/bridge_to_ethereum_config.rs

```diff
@@ -179,25 +203,144 @@ impl snowbridge_pallet_system::Config for Runtime {
 	type EthereumLocation = EthereumLocation;
 }
 
+pub struct AllowFromEthereumFrontend;
+impl Contains<Location> for AllowFromEthereumFrontend {
+	fn contains(location: &Location) -> bool {
+		match location.unpack() {
+			(1, [Parachain(para_id), PalletInstance(index)]) =>
+				*para_id == polkadot_runtime_constants::system_parachain::ASSET_HUB_ID &&
+					*index == SystemFrontendPalletInstance::get(),
+			_ => false,
+		}
+	}
+}
+
+impl snowbridge_pallet_system_v2::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type OutboundQueue = EthereumOutboundQueueV2;
+	type FrontendOrigin = EnsureXcm<AllowFromEthereumFrontend>;
+	type WeightInfo = crate::weights::snowbridge_pallet_system_v2::WeightInfo<Runtime>;
+	type GovernanceOrigin = EnsureRootWithSuccess<crate::AccountId, RootLocation>;
+	#[cfg(feature = "runtime-benchmarks")]
+	type Helper = ();
+}
+
+impl snowbridge_pallet_inbound_queue_v2::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Verifier = snowbridge_pallet_ethereum_client::Pallet<Runtime>;
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type XcmSender = xcm_config::XcmRouter;
+	#[cfg(feature = "runtime-benchmarks")]
+	type XcmSender = benchmark_helpers::DoNothingRouter;
+	type GatewayAddress = EthereumGatewayAddress;
+	#[cfg(feature = "runtime-benchmarks")]
+	type Helper = Runtime;
+	type WeightInfo = crate::weights::snowbridge_pallet_inbound_queue_v2::WeightInfo<Runtime>;
+	type AssetHubParaId = AssetHubParaId;
+	type XcmExecutor = XcmExecutor<xcm_config::XcmConfig>;
+	type MessageConverter = snowbridge_inbound_queue_primitives::v2::MessageToXcm<
+		CreateAssetCall,
+		bp_asset_hub_polkadot::CreateForeignAssetDeposit,
+		EthereumNetwork,
+		InboundQueueV2Location,
+		EthereumSystem,
+		EthereumGatewayAddress,
+		EthereumUniversalLocation,
+		AssetHubFromEthereum,
+		AssetHubUniversalLocation,
+		AccountId,
+	>;
+	type AccountToLocation = xcm_builder::AliasesIntoAccountId32<
+		xcm_config::RelayNetwork,
+		<Runtime as frame_system::Config>::AccountId,
+	>;
+	type RewardKind = BridgeReward;
+	type DefaultRewardKind = SnowbridgeReward;
+	type RewardPayment = crate::BridgeRelayers;
+}
+
+impl snowbridge_pallet_outbound_queue_v2::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Hashing = Keccak256;
+	type MessageQueue = MessageQueue;
+	// Maximum payload size for outbound messages.
+	type MaxMessagePayloadSize = ConstU32<2048>;
```

Comment:
The field `MaxMessagePayloadSize` turns out to be dead code and isn't used. Can we set this to `u32::MAX` to make that clear?

@yrong looks like we should remove `MaxMessagePayloadSize` from the pallet. I don't think it makes sense anyway. With arbitrary transact users should be able to send payloads of any size.

Reply:
Well, we still use it to benchmark the [delivery fee](https://github.com/Snowfork/runtimes/blob/73fe72538fda1eda7bd2076ecf02d7572c698263/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L729) from AH to BH. 

We can increase it somehow, but I'm not sure `u32::MAX` is appropriate, Alternatively, we might need to introduce a type like [ByteFee](https://github.com/Snowfork/runtimes/blob/73fe72538fda1eda7bd2076ecf02d7572c698263/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L594C15-L594C40), as the P-K bridge does.

Reply:
> Well, we still use it to benchmark the [delivery fee](https://github.com/Snowfork/runtimes/blob/73fe72538fda1eda7bd2076ecf02d7572c698263/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L729) from AH to BH.

Do you mean we use `MaxMessagePayloadSize` to calculate it somewhere manually? In that case, it doesn't warrant being in the runtime config.